components/kyma-environment-broker/internal/broker/instance_last_operation_test.go

```diff
@@ -56,6 +57,127 @@ func TestLastOperation_LastOperation(t *testing.T) {
 			Description: operationDescription,
 		}, response)
 	})
+	t.Run("Should convert operation's pending state to in progress", func(t *testing.T) {
+		// given
+		memoryStorage := storage.NewMemoryStorage()
+		updateOp := fixture.FixUpdatingOperation(operationID, instID)
+		updateOp.State = orchestration.Pending
+		err := memoryStorage.Operations().InsertUpdatingOperation(updateOp)
+		assert.NoError(t, err)
+
+		lastOperationEndpoint := broker.NewLastOperation(memoryStorage.Operations(), logrus.StandardLogger())
+
+		// when
+		response, err := lastOperationEndpoint.LastOperation(context.TODO(), instID, domain.PollDetails{OperationData: ""})
+		assert.Error(t, err, "instance operation with instance_id %s not found", instID)
+
+		// then
+		assert.Equal(t, domain.LastOperation{}, response)
```

Comment:
Why is it not returning the operation that has just been created?

Reply:
GetLastOperation() runs a query where operations in pending state are excluded:
https://github.com/kyma-project/control-plane/blob/main/components/kyma-environment-broker/internal/storage/postsql/read.go#L107